oak_functions/abi/src/lib.rs

```diff
@@ -83,4 +83,37 @@ extern "C" {
         inference_ptr_ptr: *mut *mut u8,
         inference_len_ptr: *mut usize,
     ) -> u32;
+
+    /// Reads a message, i.e., `dest_buf_len_ptr` bytes, from the channel with `channel_handle` into
+    /// the buffer at `dest_buf_ptr_ptr`. After a successful call to `channel_read`
```

Comment:
Note that there is no buffer at `dest_buf_ptr_ptr` when this method is invoked. Instead, that parameter should point to a variable that will receive the address of a buffer, which is allocated by the runtime by invoking the `alloc` function from the module.

Reply:
Thank you! Reformulated to make this clearer. Is it?